src/Behat/Behat/HelperContainer/Argument/ServicesResolverFactory.php

```diff
@@ -45,6 +48,8 @@ public function __construct(TaggedContainerInterface $container)
 
     /**
      * {@inheritdoc}
+     *
+     * @deprecated
```

Comment:
The deprecation should have a message giving the version it was deprecated in, and the alternative.

And I would also add a `@trigger_error('...', E_USER_DEPRECATED);` in the method.

Reply:
Good shout.